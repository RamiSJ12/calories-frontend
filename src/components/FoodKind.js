import { Grid, makeStyles, Typography } from '@material-ui/core';
import React from 'react';
import PropTypes from 'prop-types';
import commonStyles from '../utils/commonStyles';

const useStyles = makeStyles({
  autoSize: {
    maxWidth: '100%',
    maxHeight: '100%',
  },
  maxWidth50: {
    maxWidth: '50%',
  },
});

const FoodKind = ({
  sign,
  value,
  name,
  image,
}) => {
  const classes = useStyles();
  const commonClasses = commonStyles();

  return (
    <Grid className={`${commonClasses.leftMargin15} ${commonClasses.topMargin15} ${commonClasses.normalPadding} ${commonClasses.whiteBackground}`} item xs={5}>
      <Grid container justify="space-between" alignItems="center">
        <Grid xs={5} className={classes.maxWidth50} item>
          <Grid container justify="center" alignItems="center">
            <img className={classes.autoSize} src={image} alt={name} />
          </Grid>
        </Grid>
        <Grid xs={5} className={commonClasses.darkText} item>
          <Typography color="textSecondary" variant="subtitle2">{name}</Typography>
          <Typography className={commonClasses.fontWeightBold} color="textSecondary">{`${value} ${sign}`}</Typography>
        </Grid>

      </Grid>
    </Grid>
  );
};

FoodKind.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired,
  sign: PropTypes.string,
};

FoodKind.defaultProps = {
  sign: 'G',
};

export default FoodKind;
