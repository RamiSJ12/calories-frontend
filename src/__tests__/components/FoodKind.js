import renderer from 'react-test-renderer';
import FoodKind from '../../components/FoodKind';

describe('FoodKind component', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<FoodKind name="carbohydrates" value={25} image="sdkfjshfau" />);
    expect(tree).toMatchSnapshot();
  });
});
