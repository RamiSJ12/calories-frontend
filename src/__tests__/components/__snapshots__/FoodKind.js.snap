// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodKind component should match snapshot 1`] = `
<div
  className="MuiGrid-root makeStyles-leftMargin15-20 makeStyles-topMargin15-12 makeStyles-normalPadding-3 makeStyles-whiteBackground-22 MuiGrid-item MuiGrid-grid-xs-5"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
  >
    <div
      className="MuiGrid-root makeStyles-maxWidth50-2 MuiGrid-item MuiGrid-grid-xs-5"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      >
        <img
          alt="carbohydrates"
          className="makeStyles-autoSize-1"
          src="sdkfjshfau"
        />
      </div>
    </div>
    <div
      className="MuiGrid-root makeStyles-darkText-5 MuiGrid-item MuiGrid-grid-xs-5"
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
      >
        carbohydrates
      </h6>
      <p
        className="MuiTypography-root makeStyles-fontWeightBold-18 MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        25 G
      </p>
    </div>
  </div>
</div>
`;
